src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java

```diff
@@ -306,17 +310,22 @@ public boolean perform(AbstractBuild build, Launcher launcher, BuildListener lis
             writeToLog(String.format("Using App '%s'", env.expand(appArtifact)));
             Upload appUpload = adf.uploadApp(project, appArtifact);
             String appArn = appUpload.getArn();
-            String appName = String.format("%s (Jenkins)", appUpload.getName());
-
+            String deviceFarmRunName = null;
+            if (StringUtils.isBlank(runName)) {
+            	deviceFarmRunName = String.format("%s", env.get("BUILD_TAG"));
```

Comment:
We still need to have this code because runName attribute can be null if the user has upgraded the plugin but he is still using the old build instance. We have a default value for the textbox but user is free to remove it as this is an optional field